### PR TITLE
QOL-7762 Refactor configuration of file MIME types

### PR DESCRIPTION
**Pull request by ThrawnCA**
Move config into an external file so it's simpler to add more types

### CODE REVIEW

File: ckanext/qgov/common/intercepts.py [hunk 1]
```diff
@@ -48,53 +48,20 @@
...
-    'txt',
-    'wfs',
-    'wmts',
-    'xls',
-    'xlsx',
-    'xml',
-    'zip'
-]
+file_mime_config = json.load(open(
+    os.path.join(os.path.dirname(__file__), 'resources', 'resource_types.json')
```
Comment (duttonw) [inline ckanext/qgov/common/intercepts.py:52]: could we rename the file to resource_types_allowed.json ?

Comment (ThrawnCA) [reply ckanext/qgov/common/intercepts.py:52]: It's not just allowed types, it's also a list of extra entries to be added to the 'mimetypes' library.

File: .flake8 [hunk 2]
```diff
@@ -17,3 +17,4 @@ max-complexity = 10
 ignore =
     E501
     C901
+    W503
```
Comment (duttonw) [inline .flake8:20]: 👍  https://www.flake8rules.com/rules/W503.html

**Review by duttonw — APPROVED** [review]
kinda do need a file upload test for this now ;)

File: ckanext/qgov/common/test_mime_type_validation.py [hunk 3]
```diff
@@ -53,6 +54,56 @@ def test_reject_override_incompatible_prefix(self):
...
+        """ Test that resource of type 'upload' with no upload data
+        have their URL and format compared, just no content sniffing.
+        """
+        intercepts.configure({})
+        resource = {'url': 'example.csv', 'format': 'PDF'}
+        self.assertRaises(ckan.logic.ValidationError, intercepts.validate_resource_mimetype, resource)
+
+        resource['format'] = 'CSV'
+        intercepts.validate_resource_mimetype(resource)
+
```
Comment (duttonw) [inline ckanext/qgov/common/test_mime_type_validation.py:106]: 👍

**Review by duttonw — APPROVED** [review]

**Pull request merged**